openwisp-qa-check

```diff
@@ -145,12 +171,24 @@ runisort() {
 }
 
 runblack() {
-  black -S --check --diff --quiet . &&
-    echo "SUCCESS: Black check successful!" ||
-    {
-      echoerr "ERROR: Black check failed! Hint: did you forget to run openwisp-qa-format?"
-      FAILURE=1
-    }
+  if [ -n "$IGNORE_PATTERN" ]; then
+    # Convert shell pattern to regex pattern for Black
+    BLACK_EXCLUDE=$(echo "$IGNORE_PATTERN" | sed 's/\./\\./g' | sed 's/\*/.*/g')
+    
+    black -S --exclude "$BLACK_EXCLUDE" --check --diff --quiet . &&
+      echo "SUCCESS: Black check successful!" ||
+      {
+        echoerr "ERROR: Black check failed! Hint: did you forget to run openwisp-qa-format?"
+        FAILURE=1
+      }
+  else
+    black -S --check --diff --quiet . &&
+      echo "SUCCESS: Black check successful!" ||
+      {
+        echoerr "ERROR: Black check failed! Hint: did you forget to run openwisp-qa-format?"
+        FAILURE=1
+      }
+  fi
```

Comment:
here it can be refactored to remove duplicacy:

```bash
  black_args="-S --check --diff --quiet"
  
  if [ -n "$IGNORE_PATTERN" ]; then
    # Convert shell pattern to regex pattern for Black
    BLACK_EXCLUDE=$(echo "$IGNORE_PATTERN" | sed 's/\./\\./g' | sed 's/\*/.*/g')
    black_args="$black_args --exclude \"$BLACK_EXCLUDE\""
  fi

  eval "black $black_args ."
```